d3/d3.d.ts

```diff
@@ -119,7 +119,7 @@ declare module D3 {
         * @param arr Array to search
         * @param map Accsessor function
         */
-        extent<T, U>(arr: T[], map: (v: T) => U): U[];
+        extent<T, U>(arr: T[], map: (v?: T, i?: number) => U): U[];
```

Comment:
Probably want to replicate this signature for the below signatures as well.  Also, what was the conclusion about the third parameter?  I thought the javascript map signature always contained the array as the third argument?